### PR TITLE
Use stub call options for compression

### PR DESCRIPTION
**Pull request by carl-mastrangelo**
In regards to #492

### CODE REVIEW

File: stub/src/main/java/io/grpc/stub/AbstractStub.java [hunk 1]
```diff
@@ -124,6 +125,13 @@ public final S withChannel(Channel newChannel) {
   }
 
   /**
+   * Returns a new stub that uses the given compressor.
+   */
+  public final S withCompressor(Compressor c) {
```
Comment (nmittler) [inline stub/src/main/java/io/grpc/stub/AbstractStub.java]: Does this need to be experimental?

Comment (carl-mastrangelo) [inline stub/src/main/java/io/grpc/stub/AbstractStub.java]: Done.

**Comment by ejona86** [comment]
@carl-mastrangelo LGTM

**Comment by nmittler** [comment]
LGTM

**Pull request merged**